scheduler/src/cook/mesos/api.clj

```diff
@@ -1433,14 +1471,17 @@
 (defn check-retry-allowed
   [conn is-authorized-fn ctx]
   (let [request-user (get-in ctx [:request :authorization/user])
-        unauthorized-jobs (remove (partial user-can-retry-job?
-                                           conn is-authorized-fn request-user)
-                                  (::jobs ctx))]
-    (if (seq unauthorized-jobs)
-      [false {::error (->> unauthorized-jobs
-                           (map #(str "You are not authorized to retry job " % "."))
-                           (str/join \space))}]
-      true)))
+        unauthorized-job? #(not (user-can-retry-job? conn is-authorized-fn request-user %))]
+    (or (first (for [guuid (::groups ctx)
+                     :let [group (d/entity (db conn) [:group/uuid guuid])
+                           job (first (:group/job group))]
+                     :when (some-> job unauthorized-job?)]
+                 [false {::error (str "You are not authorized to retry jobs from group " guuid ".")}]))
```

Comment:
Should we add some tests for attempting to retry a group that the user is not authorized to retry?